.github/workflows/cicd-pipeline.yml

```diff
@@ -110,61 +110,64 @@ jobs:
+
+  deploy-staging-otc:
+    needs: [build-staging]
 
-      - name: Login to container repository
-        uses: docker/login-action@v1
-        with:
-          registry: ${{ env.DOCKER_REGISTRY }}
-          username: ${{ secrets.REGISTRY_USER }}
-          password: ${{ secrets.REGISTRY_PWD }}
+    if: github.event_name == 'push' && github.ref == 'refs/heads/main'
 
-      - name: Pull deploy image
-        run: docker pull ${{ env.DOCKER_REGISTRY }}/steuerlotse_deployment
-        shell: bash
+    uses: ./.github/workflows/reusable-deploy-otc.yml
+    with:
+      imageTag: ${{ github.run_number }}
+      environment: staging
+    secrets:
+      DOCKER_REGISTRY: ${{ secrets.DOCKER_REGISTRY }}
+      REGISTRY_USER: ${{ secrets.REGISTRY_USER }}
+      REGISTRY_PWD: ${{ secrets.REGISTRY_PWD }}
+      ERICA_SSH_KEY: ${{ secrets.CI_ERICA_SSH_KEY }}
+      ERICA_HOST: ${{ secrets.ERICA_HOST_OTC_STAGING }}
+      CERT_PIN: ${{ secrets.CERT_PIN_OTC_STAGING }}
+      QUEUE_URL: ${{ secrets.QUEUE_URL_OTC_STAGING }}
+      ERICA_DATABASE_URL: ${{ secrets.ERICA_DATABASE_URL_OTC_STAGING }}
+      DEPLOY_KEY: ${{ secrets.DEPLOY_KEY }}
+      ARGOCD_SERVER: ${{ secrets.ARGOCD_SERVER }}
+      ARGOCD_PIPELINE_PASSWORD: ${{ secrets.ARGOCD_PIPELINE_PASSWORD }}
+
+  track-deploy:
+    needs: [deploy-staging-syseleven, deploy-staging-otc]
 
-      - name: Run deployment
-        env:
-          NAMESPACE: erica-staging
-          IMAGE_TAG: ${{ github.run_number }}
-        run: |
-          docker run \
-          -e DOCKER_REGISTRY='${{ env.DOCKER_REGISTRY }}' \
-          -e ERICA_SSH_KEY='${{ secrets.CI_ERICA_SSH_KEY }}' \
-          -e ERICA_HOST='${{ env.ERICA_HOST }}' \
-          -e REGISTRY_PWD='${{ secrets.REGISTRY_PWD }}' \
-          -e CERT_PIN='${{ secrets.CERT_PIN_STAGING }}' \
-          -e QUEUE_URL='${{ secrets.QUEUE_URL_STAGING }}' \
-          -e ERICA_DATABASE_URL='${{ secrets.ERICA_DATABASE_URL_STAGING }}' \
-          -e KUBECONFIG_BASE64='${{ secrets.CI_KUBECONFIG_B64 }}' \
-          -e NAMESPACE='${{ env.NAMESPACE }}' \
-          -e IMAGE_TAG='${{ env.IMAGE_TAG }}' \
-          ${{ env.DOCKER_REGISTRY }}/steuerlotse_deployment erica
-        shell: bash
-
-      - name: Logout of docker container
-        run: docker logout ${{ env.DOCKER_REGISTRY }}
+    runs-on: ubuntu-latest
 
+    steps:
       - name: Report Deployment
         uses: satak/webrequest-action@v1.2.3
-        env:
-          IMAGE_TAG: ${{ github.run_number }}
         with:
           url: ${{ secrets.METRICS_DEPLOYMENT_WEBHOOK_URL }}
           method: POST
-          payload: '{"project": "Erica", "version_identifier": "${{ env.IMAGE_TAG }}", "environment": "staging", "link": "https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}"}'
+          payload: '{"project": "Erica", "version_identifier": "${{ github.run_number }}", "environment": "staging", "link": "https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}"}'
```

Comment:
Out of curiosity: why did you move the GitHub.run_number out of the env? Just because this step looks identical in most repos

Reply:
Oh just because I didn't see the value in defining a variable for a value that's used exactly once and all the other values on that line are inlined instead of variables… We should really make this a shared action though, so we can be clearer about what are parameters and what isn't